src/storage/engine/mod.rs

```diff
@@ -635,8 +639,20 @@ impl<I: Iterator> Cursor<I> {
     }
 
     #[inline]
-    pub fn valid(&self) -> bool {
-        self.iter.valid()
+    pub fn valid(&self) -> Result<bool> {
+        if !self.iter.valid() {
+            if let Err(e) = self.iter.status() {
+                CRITICAL_ERROR.with_label_values(&["rocksdb"]).inc();
```

Comment:
can we use a better name instead of "rocksdb"?

Reply:
@yiwu-arbug 

how can we meet this problem? valid is false and status returns an error.

Reply:
I am wondering if need record critical error here? Actually, it returns the same error type as `RocksDB.Write()`, but we just return the error to TiDB directly instead of record or panic.

As I understood, if valid is false and status returns an error, it indicates there is some internal problems in RocksDB. And for Titan, we will meet error when missing blob file.

Reply:
@siddontang as describe in the PR summary, if rocksdb encounter corruption or IO error, it can invalidate iterator and report error status.

Reply:
@Connor1996 not only internal error. Corruption and IO error can cause rocksdb invalidate iterator. I feel we should not panic, and just return error. Not sure how tikv used to handle rocksdb corruption and IO error.